File: services/api/app/models/domain.py
from pydantic import BaseModel


class Coordinates(BaseModel):
    lat: float
    lon: float


class GpsPoint(Coordinates):
    timestamp: str
    sogKnots: float | None = None
    cogDegrees: float | None = None


class GpsSnapshot(BaseModel):
    vesselId: str
    position: GpsPoint


class WeatherSnapshot(BaseModel):
    source: str
    timestamp: str
    windSpeedKnots: float
    windDirectionDegrees: float


class PolarPoint(BaseModel):
    twaDegrees: float
    twsKnots: float
    boatSpeedKnots: float


class PolarDiagram(BaseModel):
    vesselName: str
    points: list[PolarPoint]


class RoutePlaceholder(BaseModel):
    start: Coordinates
    end: Coordinates
    note: str

File: apps/web/src/types/domain.ts
export type Coordinates = {
  lat: number;
  lon: number;
};

export type GpsPoint = Coordinates & {
  timestamp: string;
  sogKnots?: number;
  cogDegrees?: number;
};

export type GpsSnapshot = {
  vesselId: string;
  position: GpsPoint;
};

export type WeatherSnapshot = {
  source: string;
  timestamp: string;
  windSpeedKnots: number;
  windDirectionDegrees: number;
};

export type PolarPoint = {
  twaDegrees: number;
  twsKnots: number;
  boatSpeedKnots: number;
};

export type PolarDiagram = {
  vesselName: string;
  points: PolarPoint[];
};

export type RoutePlaceholder = {
  start: Coordinates;
  end: Coordinates;
  note: string;
};
